plugins/logger/zap_adapter.go

```diff
@@ -18,18 +18,21 @@ func NewZapAdapter(zapLogger *zap.Logger) *ZapAdapter {
 	}
 }
 
-func separateFields(keyVals []interface{}) (fields []zap.Field, individualKeyVals []interface{}) {
-	for _, value := range keyVals {
-		switch value := value.(type) {
+func separateFields(keyVals []interface{}) ([]zap.Field, []interface{}) {
+	fields := []zap.Field{}
+	pairedKeyVals := []interface{}{}
```

Comment:
Please, read this guide: https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices :) I'm not a fan of various guides, but this is a very good one )